misc/quic-interop-runner/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM h2oserver/h2o-ci:ubuntu2004
+FROM h2oserver/h2o-ci:ubuntu2004 as builder
```

Comment:
This should probably be the same base image to avoid any library mismatches in the future. Let's leave this as it is for now, we do need to migrate to using an h2o docker image anyway soon.